### PR TITLE
Create and use `NumberQuestionForm` with `min` and `max` validation configurations

### PR DESCRIPTION
**Pull request by natsid**
### Description
I believe this is the last `QuestionForm` subclass we need before being able to make the `QuestionForm` class itself abstract.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Second to last piece of #589 and #590.
The last step for the above issues is to make `QuestionForm` abstract so that it can no longer be instantiated on its own.

### CODE REVIEW

File: universal-application-tool-0.0.1/app/forms/NumberQuestionForm.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    setQuestionType(QuestionType.NUMBER);
+    this.min = OptionalLong.empty();
+    this.max = OptionalLong.empty();
+  }
+
+  public NumberQuestionForm(NumberQuestionDefinition qd) {
+    super(qd);
+    setQuestionType(QuestionType.NUMBER);
+    this.min = OptionalLong.empty();
+    this.max = OptionalLong.empty();
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/forms/NumberQuestionForm.java:24]: suggestion (blocking): should these be read from the question definition?

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionConfig.java [hunk 2]
```diff
@@ -84,16 +85,19 @@ public static ContainerTag buildQuestionConfig(QuestionType type, QuestionForm q
...
+            .getContainer();
       case RADIO_BUTTON:
         return config
             .setId("single-select-question-config")
             .addMultiOptionQuestionFields((MultiOptionQuestionForm) questionForm)
             .getContainer();
-      case NUMBER:
-        return config.setId("number-question-config").addNumberQuestionConfig().getContainer();
-      case REPEATER: // fallthrough intended
+      case DROPDOWN: // fallthrough intended - no options
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/views/admin/questions/QuestionConfig.java:98]: suggestion (blocking): DROPDOWN should fall through to RADIO_BUTTON (they are both single-select questions). Would you mind adding a comment mentioning this fall through too?

File: universal-application-tool-0.0.1/app/views/components/FieldWithLabel.java [hunk 3]
```diff
@@ -171,10 +173,21 @@ public FieldWithLabel setValue(Optional<String> value) {
   public FieldWithLabel setValue(OptionalInt value) {
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/views/components/FieldWithLabel.java:173]: question: where is this method used?

**Review by carolinedanzi — APPROVED** [review]
This looks really nice! A few blocking comments but otherwise looks good to go

File: universal-application-tool-0.0.1/app/views/admin/questions/QuestionConfig.java (hunk 2 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/views/admin/questions/QuestionConfig.java:98]: Oops - thank you for catching this! I misunderstood the way switch statement fall-throughs work. Yes, a comment will help.

File: universal-application-tool-0.0.1/app/views/components/FieldWithLabel.java (hunk 3 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/views/components/FieldWithLabel.java:173]: This one is currently used for minLength and maxLength on text questions.

File: universal-application-tool-0.0.1/app/forms/NumberQuestionForm.java (hunk 1 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/forms/NumberQuestionForm.java:24]: Another great catch - thanks!

Comment (natsid) [reply universal-application-tool-0.0.1/app/forms/NumberQuestionForm.java:24]: Glad to at least see that the tests caught this mistake :)

**Comment by natsid** [comment]
Works toward #409. (There is a remaining bug with empty values that will be fixed in #805).

**Pull request merged**